modules/cdc-ext/src/main/java/org/apache/ignite/cdc/kafka/KafkaToIgniteCdcStreamer.java

```diff
@@ -0,0 +1,200 @@
+ * <pre>
+ * {@code
+ * CacheVersionConflictResolverCachePluginProvider conflictPlugin = new CacheVersionConflictResolverCachePluginProvider();
+ *
+ * conflictPlugin.setClusterId(clusterId); // Cluster id.
+ * conflictPlugin.setCaches(new HashSet<>(Arrays.asList("my-cache", "some-other-cache"))); // Caches to replicate.
+ *
+ * IgniteConfiguration cfg = ...;
+ *
+ * cfg.setPluginProviders(conflictPlugin);
+ * }
+ * </pre>
+ * Please, see {@link CacheConflictResolutionManagerImpl} for additional information.
+ *
+ * @see ChangeDataCapture
+ * @see IgniteToKafkaCdcStreamer
+ * @see ChangeDataCaptureEvent
+ * @see Applier
+ * @see CacheConflictResolutionManagerImpl
+ */
+public class KafkaToIgniteCdcStreamer implements Runnable {
+    /** Ignite configuration. */
+    private final IgniteConfiguration igniteCfg;
+
+    /** Kafka consumer properties. */
+    private final Properties kafkaProps;
+
+    /** Streamer configuration. */
+    private final KafkaToIgniteCdcStreamerConfiguration streamerCfg;
+
+    /** Executor service to run {@link Applier} instances. */
+    private final ExecutorService execSvc;
+
+    /** Appliers. */
+    private final List<Applier> appliers;
+
+    /**
+     * @param igniteCfg Ignite configuration.
+     * @param kafkaProps Kafka properties.
+     * @param streamerCfg Streamer configuration.
+     */
+    public KafkaToIgniteCdcStreamer(
+        IgniteConfiguration igniteCfg,
+        Properties kafkaProps,
+        KafkaToIgniteCdcStreamerConfiguration streamerCfg
+    ) {
+        this.igniteCfg = igniteCfg;
+        this.kafkaProps = kafkaProps;
+        this.streamerCfg = streamerCfg;
+
+        appliers = new ArrayList<>(streamerCfg.getThreadCount());
+
+        execSvc = Executors.newFixedThreadPool(streamerCfg.getThreadCount(), new ThreadFactory() {
+            private final AtomicInteger cntr = new AtomicInteger();
+
+            @Override public Thread newThread(@NotNull Runnable r) {
+                Thread th = new Thread(r);
+
+                th.setName("applier-thread-" + cntr.getAndIncrement());
+
+                return th;
+            }
+        });
+
+        if (!kafkaProps.containsKey(ConsumerConfig.GROUP_ID_CONFIG))
+            throw new IllegalArgumentException("Kafka properties don't contains " + ConsumerConfig.GROUP_ID_CONFIG);
+
+        kafkaProps.put(KEY_DESERIALIZER_CLASS_CONFIG, IntegerDeserializer.class.getName());
+        kafkaProps.put(VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class.getName());
+    }
+
+    /** {@inheritDoc} */
+    @Override public void run() {
+        try (IgniteEx ign = (IgniteEx)Ignition.start(igniteCfg)) {
+            IgniteLogger log = U.initLogger(
+                igniteCfg.getGridLogger(),
+                "kafka-ignite-streamer",
+                igniteCfg.getNodeId() != null ? igniteCfg.getNodeId() : UUID.randomUUID(),
+                igniteCfg.getWorkDirectory()
+            );
```

Comment:
duplicates code at ChangeDataCapture#ChangeDataCapture

Could we deduplicate this?